build.cmd

```diff
@@ -34,6 +35,7 @@ if not exist "%_VSCOMNTOOLS%" (
 )
 
 call "%_VSCOMNTOOLS%\VsDevCmd.bat"
+pushd %__ThisScriptDir%
```

Comment:
Is there already a matching popd? (same below)

Reply:
No there is not. But the reason for using pushd here and below is that we have new directory pushed on directory stack by VsDevCmd.bat script. On my machine it is a default source code directory set in Visual Studio settings. Using pushd to push repo root again on top of the stack restores top of stack value to one expected by build script and changes current working directory of build script to repo root. Without that change script fails to build repo.

Reply:
@BruceForstall 

According to my discussion with @weshaggard above I will work on eliminating or better controlling usage of VsDevCmd.bat script in build.cmd and run.cmd what I hope will allow to get rid of unpaired `pushd`s. For now it seems to be only solution that allows to quickly fix broken build due to VsDevCmd.bat side effects.

Reply:
I don't understand why they would ever be unpaired. It seems like you'll really mess up the user's own directory stack.

Reply:
Yeah if we cannot correctly pair it then we should probably just switch to cd instead.